### PR TITLE
Add a contribution doc on using reliable links in tooling

### PR DESCRIPTION
**Pull request by parlough**
The final piece of https://github.com/flutter/website/issues/10363 now that most existing relevant links have been updated to use the new redirects.

### CODE REVIEW

File: docs/contributing/use-reliable-links.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+Sometimes the link destination itself can add an appropriate redirect,
+but sometimes it can't and other times it's destination is not the best choice.
+
+- If you're not sure if you need a more reliable link,
+  check out [Situations to consider](#situations-to-consider).
+- If your destination does not already have a stable link,
+  and you think it would benefit from one,
+  follow the instructions in [Create a reliable link](#create-a-reliable-link).
+
+### Situations to consider
```
Comment (goderbauer) [inline docs/contributing/use-reliable-links.md:17]: Are there any situations where we shouldn't use a reliable link? I am wondering if we should have a check that disallows any non-reliable links from getting submitted.

Comment (parlough) [reply docs/contributing/use-reliable-links.md:18]: I mean I think you should always use as reliable of a link as possible, but I'm hesitant to prescribe that every link should use this format as there is still a maintenance cost. Perhaps for doc links output from the `flutter` tool or error messages.

I'd like to try this more informal, intentional process first and then after two releases or so, revisit some surrounding tooling or checks. How does that sound?

File: docs/contributing/use-reliable-links.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+> Browsers don't send URI fragments to the server,
+> so redirecting them needs extra client-side handling.
+
+
+### Create a reliable link
+
+If you need a reliable link, consider
+adding a tooling redirect to the Flutter documentation site,
+and using that in place of the intended destination.
+Tooling redirects are added a similar fashion to design-doc go links,
```
Comment (sfshaza2) [inline docs/contributing/use-reliable-links.md:44]:
```suggestion
Tooling redirects are added in a similar fashion to design-doc go links,
```

**Review by sfshaza2 — APPROVED** [review]
Other than a minor typo, lgtm

**Comment by sfshaza2** [comment]
@parlough, changing from "go" to "to" is such a simple and elegant solution. But what about all the tools that are already using "go" links? I would imagine that you _could_ set up a redirect, but that would just muddle things worse, right?

File: docs/contributing/use-reliable-links.md (hunk 1 above)
Comment (goderbauer) [reply docs/contributing/use-reliable-links.md:18]: Sounds good.

**Comment by goderbauer** [comment]
(triage): @parlough Do you still have plans for this PR?

**Comment by parlough** [comment]
> (triage): parlough Do you still have plans for this PR?

I'm going to land this as is, since the overall idea is clear and we want to stick with this approach. I'll follow up with solutions to validate usage of it.

**Pull request merged**